code/scripting/api/objs/oswpt.cpp

```diff
@@ -0,0 +1,183 @@
+//
+//
+
+#include "oswpt.h"
+#include "parse_object.h"
+#include "ship.h"
+#include "team.h"
+#include "waypoint.h"
+#include "wing.h"
+#include "ship/ship.h"
+
+namespace scripting {
+namespace api {
+
+//**********HANDLE: Wing
+ADE_OBJ(l_OSWPT, object_ship_wing_point_team, "oswpt", "OSWPT handle");
+
+ADE_FUNC(getType, l_OSWPT, nullptr, "The data-type this OSWPT yields on the get method.", "string", "The name of the data type. Either 'ship', 'parseobject' (a yet-to-spawn ship), 'wing' (can include yet-to-arrive wings with 0 current ships), 'team' (both explicit and ship-on-team), 'waypoint',  or 'none' (either explicitly specified, a ship that doesn't exist anymore, or invalid OSWPT object).")
+{
+	object_ship_wing_point_team oswpt;
+	if (!ade_get_args(L, "o", l_OSWPT.Get(&oswpt)))
+		return ade_set_error(L, "s", "none");
+
+	switch (oswpt.type) {
+	case OSWPT_TYPE_SHIP:
+		return ade_set_args(L, "s", "ship");
+	case OSWPT_TYPE_PARSE_OBJECT:
+		return ade_set_args(L, "s", "parseobject");
+	case OSWPT_TYPE_WING:
+	case OSWPT_TYPE_WING_NOT_PRESENT:
+		return ade_set_args(L, "s", "wing");
+	case OSWPT_TYPE_SHIP_ON_TEAM:
+	case OSWPT_TYPE_WHOLE_TEAM:
+		return ade_set_args(L, "s", "team");
+	case OSWPT_TYPE_WAYPOINT:
+		return ade_set_args(L, "s", "waypoint");
+	case OSWPT_TYPE_NONE:
+	case OSWPT_TYPE_EXITED:
+	default:
+		return ade_set_args(L, "s", "none");
+	}
+}
+
+ADE_FUNC(get, l_OSWPT, nullptr, "Returns the data held by this OSWPT.", "ship | parse_object | wing | team | waypoint | nil", "Returns the data held by this OSWPT, nil if type is 'none'.")
+{
+	object_ship_wing_point_team oswpt;
+	if (!ade_get_args(L, "o", l_OSWPT.Get(&oswpt)))
+		return ADE_RETURN_NIL;
+
+	switch (oswpt.type) {
+	case OSWPT_TYPE_SHIP:
+		return ade_set_args(L, "o", l_Ship.Set(object_h(oswpt.objp)));
+	case OSWPT_TYPE_PARSE_OBJECT:
+		return ade_set_args(L, "o", l_ParseObject.Set(parse_object_h(oswpt.ship_entry->p_objp)));
+	case OSWPT_TYPE_WING:
+	case OSWPT_TYPE_WING_NOT_PRESENT:
+		return ade_set_args(L, "o", l_Wing.Set(WING_INDEX(oswpt.wingp)));
+	case OSWPT_TYPE_SHIP_ON_TEAM:
```

Comment:
Will `l_Team.Set` work with the ship-on-team values?

Reply:
Yes, ship-on-team will fill the wingp property of oswpt